modules/product/src/main/java/com/opengamma/strata/product/ModelledSecurity.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.product.index.IborFutureSecurity;
+import com.opengamma.strata.product.swap.DeliverableSwapFutureSecurity;
+
+/**
+ * A security that is defined with an underlying product model.
+ * <p>
+ * This interface extends {@link Security} to provide access to the
+ * {@linkplain Product product model} underlying the security.
+ * <p>
+ * Most securities have some form of underlying model that allows pricing from first principles.
+ * For example, an {@linkplain IborFutureSecurity STIR future} contains details of the
+ * underlying Ibor index, and a {@linkplain DeliverableSwapFutureSecurity DSF} contains
+ * information on the swap that will be delivered when the future expires.
+ * This underlying model is implemented as a {@code Product}.
+ * <p>
+ * Implementations of this interface must be immutable beans.
+ */
+public interface ModelledSecurity extends Security {
```

Comment:
This name is quite a mouthful. Are there any obvious alternatives? `ModelSecurity`?